public/app/features/correlations/CorrelationsPage.test.tsx

```diff
@@ -218,6 +229,25 @@ const renderWithContext = async (
   return renderResult;
 };
 
+const mockQueryRunner = () => {
+  let emitter: { subscriber: Subscriber<PanelData> | undefined } = { subscriber: undefined };
+  const observable: Observable<PanelData> = new Observable((subscriber) => {
+    emitter.subscriber = subscriber;
+  });
+  const factory: QueryRunnerFactory = () => ({
+    get: () => observable,
+    run: () => {},
+    cancel: () => {},
+    destroy: () => {},
+  });
+
+  setQueryRunnerFactory(factory);
+
+  return emitter;
+};
+
+let emit: { subscriber: Subscriber<PanelData> | undefined } = { subscriber: undefined };
+
```

Comment:
what about:
```suggestion
type NextFn = (value?: PanelData) => void;
const mockQueryRunner = () => {
  let next: NextFn = () => {};
  const observable: Observable<PanelData> = new Observable((subscriber) => {
    next = subscriber.next;
  });
  const factory: QueryRunnerFactory = () => ({
    get: () => observable,
    run: () => {},
    cancel: () => {},
    destroy: () => {},
  });

  setQueryRunnerFactory(factory);

  return next;
};

let emit: NextFn = () => {};
```

and use it like

```ts
emit({ series: [], timeRange: getDefaultTimeRange(), state: LoadingState.Error });
```

so we don't have to do ugly non-null assertion? 😛

Reply:
I'd love to accept your suggestion @Elfo404 but somehow it suppresses the label for the validation and therefore the test fails. Do you have any idea why this happens?
<img width="1044" alt="Screenshot 2022-11-28 at 16 15 23" src="https://user-images.githubusercontent.com/48948963/204313953-7da3271c-1b14-4f6d-a1a9-8eea53b4d259.png">

Reply:
Applied suggested changes from the following thread: https://raintank-corp.slack.com/archives/C02TXB92G6T/p1669656594428139

Reply:
Dropped the suggestion. See thread in Slack (link above)